rclcpp/include/rclcpp/any_subscription_callback.hpp

```diff
@@ -406,10 +406,7 @@ class AnySubscriptionCallback
 
   /// Function for shared_ptr to non-const MessageT, which is deprecated.
   template<typename SetT>
-  // TODO(wjwwood): enable this deprecation after Galactic
-  // [[deprecated(
-  //   "use 'void (std::shared_ptr<const MessageT>)' instead"
-  // )]]
+  [[deprecated("use 'void(std::shared_ptr<const MessageT>)' instead")]]
```

Comment:
Based on an offline discussion with @aprotyas:

We are having trouble suppressing these deprecations in one of the tests, so I suggest doing this as a workaround:

```suggestion
#if !defined(RCLCPP_AVOID_DEPRECATIONS_FOR_TESTS)
  [[deprecated("use 'void(std::shared_ptr<const MessageT>)' instead")]]
#endif
```

And then in the test file:

```c++
#define RCLCPP_AVOID_DEPRECATIONS_FOR_TESTS 1
#include "rclcpp/rclcpp.hpp"
```

Reply:
0e66df0 introduces the changes outlined in this comment.